src/target/riscv/riscv.c

```diff
@@ -4718,6 +4718,10 @@ unsigned int riscv_count_harts(struct target *target)
  */
 static bool gdb_regno_cacheable(enum gdb_regno regno, bool is_write)
 {
+	/* Writes to register zero are rare and should always result in 0 */
```

Comment:
If you settle on the solution of making x0 non-cacheable, I'd just use a different comment:

```suggestion
	/* Don't cache values of x0 -- this prevents storing non-zero (bogus) values in the cache. */
```

Reply:
Done